task_2/task_2.txt

```diff
@@ -0,0 +1,23 @@
+There are several possible ways to provide the secrets to running pods securely:
+
+1. AWS KMS envelope encryption https://aws.amazon.com/blogs/containers/using-eks-encryption-provider-support-for-defense-in-depth/;
++ completely transparent to the application;
++ does not break common flow, secrets are referenced as usual;
++ no additional configuration is needed;
+
+2. kubernetes-external-secrets which creates ExternalSecrets crd, pulling them from AWS Parameter Store or SecretsManager https://github.com/godaddy/kubernetes-external-secrets 
```

Comment:
We love external-secrets and used it a few times in our projects. Can you describe how would you implement a keys rotation procedure on-top of external-secrets? I mean, what if you'll need to rotate a specific set of keys every month. How these changes are going to be propagated to the running applications inside EKS?

Reply:
It could be setup as periodic job which does the following:
1. Generate new secret version in AWS SecretsManager
2. Obtains a list of all pods referencing this secret
3. Triggers restart of all pods from 2.

Reply:
@eugenetaranov how about the following approach: https://github.com/godaddy/kubernetes-external-secrets/issues/38 ?

Reply:
reloader looks like a way to go, thank you!